### ISSUE TITLE
Android PluginResult intermittently not being forwarded to JS

### ISSUE THREAD

**Issue by djjolicoeur** [AppGyver/steroids#872]
I originally started looking into this here: https://github.com/AppGyver/PushNotifications/issues/19 , but as I dug into the issue, it became clear that the issue was deeper.  Occasionally, callbacks will make all the way through the plugin, and by that I mean `CallbackContext.sendPluginResult()` is being successfully fired, but the corresponding javascript never makes it through the `NativeToJSMessageQueue` on the `AGWebView`.  I suspect this has something to do with the coordination of the `NativeToJSMessageQueue` and  the `BridgeMode` on `AGWebView`, but to my knowledge, that code is AppGyver's and I do not have access to dig deeper.   If I am incorrect on that, please point me in the direction of that code and I will happily dig away.  For more information please see my comments on https://github.com/AppGyver/PushNotifications/issues/19 , this is a major blocker for us as we don't think its appropriate to push out an app to our customers that  exhibits non-deterministic functionality of the features we rely on.  I am more than willing to help out wherever/however I can.

**Comment by harsaharsa** [comment]
Can you describe your app environment in more detail? How many WebViews there are in your app? Are you receiving push notifications in one WebView or several? Are you experiencing this problem only on Crosswalk, or have you tried the platform WebView build?

**Comment by djjolicoeur** [comment]
We are targeting crosswalk, and our app has a total of 2 WebViews, only one of which interacts with the PushNotification plugin in any way.  I think the logs I posted on the other ticket do the situation we are seeing justice, and if you want to verify this behavior, you can use my PR of the plugin, https://github.com/AppGyver/PushNotifications/pull/18 , which has logging all the way through.  As I  mentioned, this is non-deterministic.  sometimes it works, sometimes it does not, but there are no errors (BTW my PR implements the error callback for registration, which is not implemented in the plugin, currently).  I think it is important to note that at least one other user commented on the other ticket saying that they are still seeing the same intermittent failures with a different plugin, so this problem is not just isolated to our app.

**Comment by djjolicoeur** [comment]
I tested on the latest edge12 and I am still seeing the issue persist.  I captured some log output to help track down what is happening here:

1) the callback did not fire on device.  I noticed the extra logging is showing that the message _was_ flushed from the queue.  could there be a condition in the CordovaInterface that would cause the message to not make it all the way to the javascript?  

```
V/PushPlugin( 6060): handleRegister -> data: [{"senderID":"REDACTED","sound":"true","alert":"true","badge":"true"}]
V/PushPlugin( 6060): PushPlugin == com.plugin.gcm.PushPlugin@42568e90
D/CordovaPlugin( 6060): Sending plugin result to WebView: "No result"
D/AGWebView( 6060): Added plugin result to JS message queue: "No result"
V/PushPlugin( 6060): iFace = com.plugin.gcm.PushPlugin@42568e90
V/PushPlugin( 6060): Exiting exec
D/GCMRegistrar( 6060): resetting backoff for com.vitalreactor.orchestra2
V/GCMRegistrar( 6060): Registering app com.vitalreactor.orchestra2 of senders REDACTED
D/DisplayPowerController(  487): [DAB] updateAutoBrightnessSEC : 20(20.0)    0.0 < 5.0 < 15.0 (0.0)
I/DisplayPowerController(  487): [DAB] fileWriteInt : /sys/class/backlight/panel/auto_brightness  value : 1
V/GCMBroadcastReceiver( 6060): onReceive: com.google.android.c2dm.intent.REGISTRATION
V/GCMRegistrar( 6060): Setting the name of retry receiver class to com.plugin.gcm.CordovaGCMBroadcastReceiver
V/GCMBroadcastReceiver( 6060): GCM IntentService class: com.plugin.gcm.GCMIntentService
V/GCMBaseIntentService( 6060): Acquiring wakelock
D/DisplayPowerController(  487): animation target = 20 (PSM:false, AB limit:(-1 ~ -1) MB Limit:(-1 ~ -1) MAdj:(0, (0, 255)))
D/lights  (  487): lcd : 90 +
D/lights  (  487): lcd : 90 -
V/GCMBaseIntentService( 6060): Intent service name: GCMIntentService-GCMIntentService-1
V/GCMRegistrar( 6060): Registering receiver
D/GCMBaseIntentService( 6060): handleRegistration: registrationId = APA91bH6gBEygz-CnRBm84p4ldcLfBFXyV7Yye5TS_vuQJ2ARLQEBszMx3BXCGc7PQSWunvy-BhYCz47B7b3Gi3o9LOHaPo8ND3oT0O9HdJV5_GOhKRy7i\
8rKlq9PqsTNMpCdeGXxRAFDf1XfULpmKjC1gN_nYqeTQ, error = null, unregistered = null
D/GCMRegistrar( 6060): resetting backoff for com.vitalreactor.orchestra2
V/GCMRegistrar( 6060): Saving regId on app version 195
D/PushPlugin-GCMIntentService( 6060): onRegistered: APA91bH6gBEygz-CnRBm84p4ldcLfBFXyV7Yye5TS_vuQJ2ARLQEBszMx3BXCGc7PQSWunvy-BhYCz47B7b3Gi3o9LOHaPo8ND3oT0O9HdJV5_GOhKRy7i8rKlq9PqsTNMpCde\
GXxRAFDf1XfULpmKjC1gN_nYqeTQ
V/PushPlugin-NotificationService( 6060): GCM Registration webview: AGWebView -> URL: http://localhost/index.html uuid: 8aa8e6ac-1e4e-4e71-986b-e3ba4bd014a7
V/PushPlugin-NotificationService( 6060): handler found, sending registration id APA91bH6gBEygz-CnRBm84p4ldcLfBFXyV7Yye5TS_vuQJ2ARLQEBszMx3BXCGc7PQSWunvy-BhYCz47B7b3Gi3o9LOHaPo8ND3oT0O9Hd\
JV5_GOhKRy7i8rKlq9PqsTNMpCdeGXxRAFDf1XfULpmKjC1gN_nYqeTQto handler: com.plugin.gcm.PushPlugin@42568e90
V/PushPlugin( 6060): Registration Success called: APA91bH6gBEygz-CnRBm84p4ldcLfBFXyV7Yye5TS_vuQJ2ARLQEBszMx3BXCGc7PQSWunvy-BhYCz47B7b3Gi3o9LOHaPo8ND3oT0O9HdJV5_GOhKRy7i8rKlq9PqsTNMpCdeGX\
xRAFDf1XfULpmKjC1gN_nYqeTQ for instance WebView AGWebView -> URL: http://localhost/index.html uuid: 8aa8e6ac-1e4e-4e71-986b-e3ba4bd014a7 callBack PushPlugin1175916636
D/CordovaPlugin( 6060): Sending plugin result to WebView: "APA91bH6gBEygz-CnRBm84p4ldcLfBFXyV7Yye5TS_vuQJ2ARLQEBszMx3BXCGc7PQSWunvy-BhYCz47B7b3Gi3o9LOHaPo8ND3oT0O9HdJV5_GOhKRy7i8rKlq9Pqs\
TNMpCdeGXxRAFDf1XfULpmKjC1gN_nYqeTQ"
D/AGWebView( 6060): Added plugin result to JS message queue: "APA91bH6gBEygz-CnRBm84p4ldcLfBFXyV7Yye5TS_vuQJ2ARLQEBszMx3BXCGc7PQSWunvy-BhYCz47B7b3Gi3o9LOHaPo8ND3oT0O9HdJV5_GOhKRy7i8rKlq9\
PqsTNMpCdeGXxRAFDf1XfULpmKjC1gN_nYqeTQ"
D/JsMessageQueue( 6060): Enqueued JsMessage: "APA91bH6gBEygz-CnRBm84p4ldcLfBFXyV7Yye5TS_vuQJ2ARLQEBszMx3BXCGc7PQSWunvy-BhYCz47B7b3Gi3o9LOHaPo8ND3oT0O9HdJV5_GOhKRy7i8rKlq9PqsTNMpCdeGXxRAF\
Df1XfULpmKjC1gN_nYqeTQ", paused: false
V/GCMBaseIntentService( 6060): Releasing wakelock
D/JsMessageQueue( 6060): Flushed 1 messages to JS side.
D/SensorService(  487):   -1.1 -0.2 9.0
... no callback ever fired
```

2) the callback _did_ fire on device.  I did notice that the actual device was much more reliable on this build, with regard to the callback reaching javascript.  As demonstrated above, however, the problem does still persist just not at the same frequency as before.  

```
V/GCMBroadcastReceiver( 5226): onReceive: com.google.android.c2dm.intent.REGISTRATION
V/GCMRegistrar( 5226): Setting the name of retry receiver class to com.plugin.gcm.CordovaGCMBroadcastReceiver
V/GCMBroadcastReceiver( 5226): GCM IntentService class: com.plugin.gcm.GCMIntentService
V/GCMBaseIntentService( 5226): Acquiring wakelock
V/GCMBaseIntentService( 5226): Intent service name: GCMIntentService-GCMIntentService-1
V/GCMRegistrar( 5226): Registering receiver
D/GCMBaseIntentService( 5226): handleRegistration: registrationId = APA91bF5TnIKGHWtgcbkDJrMep9N_vELMDvnXHFrqJzRAOUaFbb9KEvSnrvWdTjV9Hc78dYh_vEvKrui41sI8VckxGXj4MRvGntocTpHw2_ZOw0UFyI3cw\
Zr5g6-iJO7zHn8KQ8oDTLZ67_mrhFt-5cqnTK2-PHBEQ, error = null, unregistered = null
D/GCMRegistrar( 5226): resetting backoff for com.vitalreactor.orchestra2
V/GCMRegistrar( 5226): Saving regId on app version 195
D/PushPlugin-GCMIntentService( 5226): onRegistered: APA91bF5TnIKGHWtgcbkDJrMep9N_vELMDvnXHFrqJzRAOUaFbb9KEvSnrvWdTjV9Hc78dYh_vEvKrui41sI8VckxGXj4MRvGntocTpHw2_ZOw0UFyI3cwZr5g6-iJO7zHn8KQ\
8oDTLZ67_mrhFt-5cqnTK2-PHBEQ
V/PushPlugin-NotificationService( 5226): GCM Registration webview: AGWebView -> URL: http://localhost/index.html uuid: e7e2fc8d-f7bb-4da8-8b74-1e47a49a0dc8
V/PushPlugin-NotificationService( 5226): handler found, sending registration id APA91bF5TnIKGHWtgcbkDJrMep9N_vELMDvnXHFrqJzRAOUaFbb9KEvSnrvWdTjV9Hc78dYh_vEvKrui41sI8VckxGXj4MRvGntocTpHw2\
_ZOw0UFyI3cwZr5g6-iJO7zHn8KQ8oDTLZ67_mrhFt-5cqnTK2-PHBEQto handler: com.plugin.gcm.PushPlugin@42563fb8
V/PushPlugin( 5226): Registration Success called: APA91bF5TnIKGHWtgcbkDJrMep9N_vELMDvnXHFrqJzRAOUaFbb9KEvSnrvWdTjV9Hc78dYh_vEvKrui41sI8VckxGXj4MRvGntocTpHw2_ZOw0UFyI3cwZr5g6-iJO7zHn8KQ8o\
DTLZ67_mrhFt-5cqnTK2-PHBEQ for instance WebView AGWebView -> URL: http://localhost/index.html uuid: e7e2fc8d-f7bb-4da8-8b74-1e47a49a0dc8 callBack PushPlugin72879178
D/CordovaPlugin( 5226): Sending plugin result to WebView: "APA91bF5TnIKGHWtgcbkDJrMep9N_vELMDvnXHFrqJzRAOUaFbb9KEvSnrvWdTjV9Hc78dYh_vEvKrui41sI8VckxGXj4MRvGntocTpHw2_ZOw0UFyI3cwZr5g6-iJO\
7zHn8KQ8oDTLZ67_mrhFt-5cqnTK2-PHBEQ"
D/AGWebView( 5226): Added plugin result to JS message queue: "APA91bF5TnIKGHWtgcbkDJrMep9N_vELMDvnXHFrqJzRAOUaFbb9KEvSnrvWdTjV9Hc78dYh_vEvKrui41sI8VckxGXj4MRvGntocTpHw2_ZOw0UFyI3cwZr5g6-\
iJO7zHn8KQ8oDTLZ67_mrhFt-5cqnTK2-PHBEQ"
D/JsMessageQueue( 5226): Enqueued JsMessage: "APA91bF5TnIKGHWtgcbkDJrMep9N_vELMDvnXHFrqJzRAOUaFbb9KEvSnrvWdTjV9Hc78dYh_vEvKrui41sI8VckxGXj4MRvGntocTpHw2_ZOw0UFyI3cwZr5g6-iJO7zHn8KQ8oDTLZ\
67_mrhFt-5cqnTK2-PHBEQ", paused: false
V/GCMBaseIntentService( 5226): Releasing wakelock
D/JsMessageQueue( 5226): Flushed 1 messages to JS side.
I/chromium( 5226): [INFO:CONSOLE(2257)] "in success-handler: ", source: http://localhost/javascripts/js/orchestra.js (2257)
I/chromium( 5226): [INFO:CONSOLE(2254)] "On configure plugin: ", source: http://localhost/javascripts/js/orchestra.js (2254)
```

3) callback failed from emulator (less noise). it almost universally fails from the emulator which would lead me to believe there is some kind of timing error that only gets triggered occasionally on a device.  I would say the callback falls to fire on an emulator about 90% of the time, which may be useful for tracking down the bug more reliably.  

```
D/ExposedJsApi( 2797): Execing PushPlugin::register
V/PushPlugin( 2797): handleRegister -> data: [{"senderID":"REDACTED","sound":"true","alert":"true","badge":"true"}]
V/PushPlugin( 2797): PushPlugin == com.plugin.gcm.PushPlugin@b12c89f0
D/CordovaPlugin( 2797): Sending plugin result to WebView: "No result"
D/AGWebView( 2797): Added plugin result to JS message queue: "No result"
V/PushPlugin( 2797): iFace = com.plugin.gcm.PushPlugin@b12c89f0
V/PushPlugin( 2797): Exiting exec
D/GCMRegistrar( 2797): resetting backoff for com.vitalreactor.orchestra2
V/GCMRegistrar( 2797): Registering app com.vitalreactor.orchestra2 of senders REDACTED
D/dalvikvm( 1666): GC_CONCURRENT freed 424K, 22% free 4189K/5360K, paused 1ms+0ms, total 29ms
V/GCMBroadcastReceiver( 2797): onReceive: com.google.android.c2dm.intent.REGISTRATION
V/GCMRegistrar( 2797): Setting the name of retry receiver class to com.plugin.gcm.CordovaGCMBroadcastReceiver
V/GCMBroadcastReceiver( 2797): GCM IntentService class: com.plugin.gcm.GCMIntentService
V/GCMBaseIntentService( 2797): Acquiring wakelock
V/GCMBaseIntentService( 2797): Intent service name: GCMIntentService-GCMIntentService-1
V/GCMRegistrar( 2797): Registering receiver
D/GCMBaseIntentService( 2797): handleRegistration: registrationId = APA91bEbxW5J0-KyAdI5bdRFpWsxGcpctTCIHv-apwm4GYXFY9CYkYXN6yRGUKmOCF9-FOqd3b8THD_eSeloBxUTdL9Rbgj2xIScmlLoYo_1wk87hmbH4j\
kc4EqGozCqJ7ZWkmu-b_29UH3WBbvcQwU-hyW1Gf-Eug, error = null, unregistered = null
D/GCMRegistrar( 2797): resetting backoff for com.vitalreactor.orchestra2
V/GCMRegistrar( 2797): Saving regId on app version 195
D/PushPlugin-GCMIntentService( 2797): onRegistered: APA91bEbxW5J0-KyAdI5bdRFpWsxGcpctTCIHv-apwm4GYXFY9CYkYXN6yRGUKmOCF9-FOqd3b8THD_eSeloBxUTdL9Rbgj2xIScmlLoYo_1wk87hmbH4jkc4EqGozCqJ7ZWkm\
u-b_29UH3WBbvcQwU-hyW1Gf-Eug
D/dalvikvm( 2797): GC_CONCURRENT freed 860K, 14% free 6212K/7204K, paused 15ms+0ms, total 18ms
V/PushPlugin-NotificationService( 2797): GCM Registration webview: AGWebView -> URL: http://localhost/index.html uuid: cf2d937f-079f-4abe-bd1f-566ca0a3fffe
V/PushPlugin-NotificationService( 2797): handler found, sending registration id APA91bEbxW5J0-KyAdI5bdRFpWsxGcpctTCIHv-apwm4GYXFY9CYkYXN6yRGUKmOCF9-FOqd3b8THD_eSeloBxUTdL9Rbgj2xIScmlLoYo\
_1wk87hmbH4jkc4EqGozCqJ7ZWkmu-b_29UH3WBbvcQwU-hyW1Gf-Eugto handler: com.plugin.gcm.PushPlugin@b12c89f0
V/PushPlugin( 2797): Registration Success called: APA91bEbxW5J0-KyAdI5bdRFpWsxGcpctTCIHv-apwm4GYXFY9CYkYXN6yRGUKmOCF9-FOqd3b8THD_eSeloBxUTdL9Rbgj2xIScmlLoYo_1wk87hmbH4jkc4EqGozCqJ7ZWkmu-\
b_29UH3WBbvcQwU-hyW1Gf-Eug for instance WebView AGWebView -> URL: http://localhost/index.html uuid: cf2d937f-079f-4abe-bd1f-566ca0a3fffe callBack PushPlugin301889125
D/CordovaPlugin( 2797): Sending plugin result to WebView: "APA91bEbxW5J0-KyAdI5bdRFpWsxGcpctTCIHv-apwm4GYXFY9CYkYXN6yRGUKmOCF9-FOqd3b8THD_eSeloBxUTdL9Rbgj2xIScmlLoYo_1wk87hmbH4jkc4EqGozC\
qJ7ZWkmu-b_29UH3WBbvcQwU-hyW1Gf-Eug"
D/AGWebView( 2797): Added plugin result to JS message queue: "APA91bEbxW5J0-KyAdI5bdRFpWsxGcpctTCIHv-apwm4GYXFY9CYkYXN6yRGUKmOCF9-FOqd3b8THD_eSeloBxUTdL9Rbgj2xIScmlLoYo_1wk87hmbH4jkc4EqG\
ozCqJ7ZWkmu-b_29UH3WBbvcQwU-hyW1Gf-Eug"
D/JsMessageQueue( 2797): Enqueued JsMessage: "APA91bEbxW5J0-KyAdI5bdRFpWsxGcpctTCIHv-apwm4GYXFY9CYkYXN6yRGUKmOCF9-FOqd3b8THD_eSeloBxUTdL9Rbgj2xIScmlLoYo_1wk87hmbH4jkc4EqGozCqJ7ZWkmu-b_29\
UH3WBbvcQwU-hyW1Gf-Eug", paused: false
V/GCMBaseIntentService( 2797): Releasing wakelock
D/JsMessageQueue( 2797): Flushed 1 messages to JS side.
... no callback ever fired
```

**Comment by harsaharsa** [comment]
Very strange. I still haven't been able to reproduce this on my end, so I'm a bit at loss here. :/ One more logging thing you could do is grab the `cordova.js` that is being served at `http://localhost/cordova.js` (e.g. via the Chrome Inspector), copy it to your project, rename it to e.g. `my_awesome_cordova.js` and load that instead of the bundled version. Then, you can find the native `exec` code at line 852 onwards and add some `console.log` there – to see if the `"Flushed 1 messages to JS side"` event actually triggers the JS side to fetch new stuff from the native message queue, if the messages are visible etc.

Just note that for the logs to show up in ADB properly, you need to pass a single string – `console.log("one","two");` will just output `"one"`.

Also, are you 100% sure that the callback function is being registered correctly? Can you reproduce the callback-not-firing with any other API?

**Comment by djjolicoeur** [comment]
If the callback wasn't being registered, the PluginResult would have nothing to get sent to the WebView from, as that is triggered from the CallbackContext. 

I injected a custom `cordova.js` with additional logging to trace messages from native in our build to gain some insight into what is happening on the JS side, and it would appear that the failed callbacks are never making it to javascript.

1) succesful callback
``` 
D/ExposedJsApi( 4267): Execing PushPlugin::register
V/PushPlugin( 4267): handleRegister -> data: [{"senderID":"REDACTED","sound":"true","alert":"true","badge":"true"}]
V/PushPlugin( 4267): PushPlugin == com.plugin.gcm.PushPlugin@b13701f8
D/CordovaPlugin( 4267): Sending plugin result to WebView: "No result"
D/AGWebView( 4267): Added plugin result to JS message queue: "No result"
V/PushPlugin( 4267): iFace = com.plugin.gcm.PushPlugin@b13701f8
V/PushPlugin( 4267): Exiting exec
D/GCMRegistrar( 4267): resetting backoff for com.vitalreactor.orchestra2
V/GCMRegistrar( 4267): Registering app com.vitalreactor.orchestra2 of senders REDACTED
D/dalvikvm( 1794): GC_CONCURRENT freed 381K, 13% free 3945K/4532K, paused 0ms+0ms, total 3ms
V/GCMBroadcastReceiver( 4267): onReceive: com.google.android.c2dm.intent.REGISTRATION
V/GCMRegistrar( 4267): Setting the name of retry receiver class to com.plugin.gcm.CordovaGCMBroadcastReceiver
V/GCMBroadcastReceiver( 4267): GCM IntentService class: com.plugin.gcm.GCMIntentService
V/GCMBaseIntentService( 4267): Acquiring wakelock
V/GCMBaseIntentService( 4267): Intent service name: GCMIntentService-GCMIntentService-1
V/GCMRegistrar( 4267): Registering receiver
D/GCMBaseIntentService( 4267): handleRegistration: registrationId = APA91bHxtQ-L9KKvx1rxI8g7VliWgj8ER1dMIfeOqU4f7Lq-TFhyuOw-04OOsr6583jTICWCQNcKUjS1A5qKTSHHQ8NRBsRqRlSmB9gh2UTEutkBOkN9mR\
f-3QehaBXogVARar9GhtEBUF_8tlpl8uYfhKKk5onakA, error = null, unregistered = null
D/GCMRegistrar( 4267): resetting backoff for com.vitalreactor.orchestra2
V/GCMRegistrar( 4267): Saving regId on app version 195
D/PushPlugin-GCMIntentService( 4267): onRegistered: APA91bHxtQ-L9KKvx1rxI8g7VliWgj8ER1dMIfeOqU4f7Lq-TFhyuOw-04OOsr6583jTICWCQNcKUjS1A5qKTSHHQ8NRBsRqRlSmB9gh2UTEutkBOkN9mRf-3QehaBXogVARar\
9GhtEBUF_8tlpl8uYfhKKk5onakA
V/PushPlugin-NotificationService( 4267): GCM Registration webview: AGWebView -> URL: http://localhost/index.html uuid: 0fe4734c-8ab5-4335-82ac-eba3efd28a07
V/PushPlugin-NotificationService( 4267): handler found, sending registration id APA91bHxtQ-L9KKvx1rxI8g7VliWgj8ER1dMIfeOqU4f7Lq-TFhyuOw-04OOsr6583jTICWCQNcKUjS1A5qKTSHHQ8NRBsRqRlSmB9gh2U\
TEutkBOkN9mRf-3QehaBXogVARar9GhtEBUF_8tlpl8uYfhKKk5onakAto handler: com.plugin.gcm.PushPlugin@b13701f8
V/PushPlugin( 4267): Registration Success called: APA91bHxtQ-L9KKvx1rxI8g7VliWgj8ER1dMIfeOqU4f7Lq-TFhyuOw-04OOsr6583jTICWCQNcKUjS1A5qKTSHHQ8NRBsRqRlSmB9gh2UTEutkBOkN9mRf-3QehaBXogVARar9G\
htEBUF_8tlpl8uYfhKKk5onakA for instance WebView AGWebView -> URL: http://localhost/index.html uuid: 0fe4734c-8ab5-4335-82ac-eba3efd28a07 callBack PushPlugin43403542
D/CordovaPlugin( 4267): Sending plugin result to WebView: "APA91bHxtQ-L9KKvx1rxI8g7VliWgj8ER1dMIfeOqU4f7Lq-TFhyuOw-04OOsr6583jTICWCQNcKUjS1A5qKTSHHQ8NRBsRqRlSmB9gh2UTEutkBOkN9mRf-3QehaBX\
ogVARar9GhtEBUF_8tlpl8uYfhKKk5onakA"
D/AGWebView( 4267): Added plugin result to JS message queue: "APA91bHxtQ-L9KKvx1rxI8g7VliWgj8ER1dMIfeOqU4f7Lq-TFhyuOw-04OOsr6583jTICWCQNcKUjS1A5qKTSHHQ8NRBsRqRlSmB9gh2UTEutkBOkN9mRf-3Qeh\
aBXogVARar9GhtEBUF_8tlpl8uYfhKKk5onakA"
D/JsMessageQueue( 4267): Enqueued JsMessage: "APA91bHxtQ-L9KKvx1rxI8g7VliWgj8ER1dMIfeOqU4f7Lq-TFhyuOw-04OOsr6583jTICWCQNcKUjS1A5qKTSHHQ8NRBsRqRlSmB9gh2UTEutkBOkN9mRf-3QehaBXogVARar9GhtEB\
UF_8tlpl8uYfhKKk5onakA", paused: false
D/JsMessageQueue( 4267): Flushed 1 messages to JS side.
V/GCMBaseIntentService( 4267): Releasing wakelock
I/chromium( 4267): [INFO:CONSOLE(1057)] "PROCESS MESSAGES", source: http://localhost/test-cordova.js (1057)
I/chromium( 4267): [INFO:CONSOLE(1073)] "POPPED MESSAGE FROM QUEUE "S01 PushPlugin43403542 sAPA91bHxtQ-L9KKvx1rxI8g7VliWgj8ER1dMIfeOqU4f7Lq-TFhyuOw-04OOsr6583jTICWCQNcKUjS1A5qKTSHHQ8NRBs\
RqRlSmB9gh2UTEutkBOkN9mRf-3QehaBXogVARar9GhtEBUF_8tlpl8uYfhKKk5onakA"", source: http://localhost/test-cordova.js (1073)
I/chromium( 4267): [INFO:CONSOLE(1003)] "MESSAGE RECEIVED "S01 PushPlugin43403542 sAPA91bHxtQ-L9KKvx1rxI8g7VliWgj8ER1dMIfeOqU4f7Lq-TFhyuOw-04OOsr6583jTICWCQNcKUjS1A5qKTSHHQ8NRBsRqRlSmB9g\
h2UTEutkBOkN9mRf-3QehaBXogVARar9GhtEBUF_8tlpl8uYfhKKk5onakA"", source: http://localhost/test-cordova.js (1003)
I/chromium( 4267): [INFO:CONSOLE(1016)] "PAYLOAD KIND s", source: http://localhost/test-cordova.js (1016)
```

2) Failure
```
D/ExposedJsApi( 3847): Execing PushPlugin::register
V/PushPlugin( 3847): handleRegister -> data: [{"senderID":"REDACTED","sound":"true","alert":"true","badge":"true"}]
V/PushPlugin( 3847): PushPlugin == com.plugin.gcm.PushPlugin@b1312a30
D/CordovaPlugin( 3847): Sending plugin result to WebView: "No result"
D/AGWebView( 3847): Added plugin result to JS message queue: "No result"
V/PushPlugin( 3847): iFace = com.plugin.gcm.PushPlugin@b1312a30
V/PushPlugin( 3847): Exiting exec
D/GCMRegistrar( 3847): resetting backoff for com.vitalreactor.orchestra2
I/chromium( 3847): [INFO:CONSOLE(1057)] "PROCESS MESSAGES", source: http://localhost/test-cordova.js (1057)
V/GCMRegistrar( 3847): Registering app com.vitalreactor.orchestra2 of senders REDACTED
I/chromium( 3847): [INFO:CONSOLE(1073)] "POPPED MESSAGE FROM QUEUE "S11 NetworkStatus27794449 s3g"", source: http://localhost/test-cordova.js (1073)
I/chromium( 3847): [INFO:CONSOLE(1003)] "MESSAGE RECEIVED "S11 NetworkStatus27794449 s3g"", source: http://localhost/test-cordova.js (1003)
I/chromium( 3847): [INFO:CONSOLE(1016)] "PAYLOAD KIND s", source: http://localhost/test-cordova.js (1016)
D/dalvikvm( 1759): GC_FOR_ALLOC freed 176K, 13% free 3947K/4532K, paused 3ms, total 3ms
V/GCMBroadcastReceiver( 3847): onReceive: com.google.android.c2dm.intent.REGISTRATION
V/GCMRegistrar( 3847): Setting the name of retry receiver class to com.plugin.gcm.CordovaGCMBroadcastReceiver
V/GCMBroadcastReceiver( 3847): GCM IntentService class: com.plugin.gcm.GCMIntentService
V/GCMBaseIntentService( 3847): Acquiring wakelock
V/GCMBaseIntentService( 3847): Intent service name: GCMIntentService-GCMIntentService-1
V/GCMRegistrar( 3847): Registering receiver
D/GCMBaseIntentService( 3847): handleRegistration: registrationId = APA91bGN7FCQ2dfjMR757yAtn6fLiuVM0IeZmzwJKCUe6aNk9v7pIMXOXB16FPZFzME1r0ZrdPQYlpxLW8QzR2YepN2kv8WOiDW-kqLrmXXjiaj-Dqobjy\
zskFJzYQPZCMPSA-vPb1uEpjb7HXSY4OP8UZiNkvw5kg, error = null, unregistered = null
D/GCMRegistrar( 3847): resetting backoff for com.vitalreactor.orchestra2
V/GCMRegistrar( 3847): Saving regId on app version 195
D/PushPlugin-GCMIntentService( 3847): onRegistered: APA91bGN7FCQ2dfjMR757yAtn6fLiuVM0IeZmzwJKCUe6aNk9v7pIMXOXB16FPZFzME1r0ZrdPQYlpxLW8QzR2YepN2kv8WOiDW-kqLrmXXjiaj-DqobjyzskFJzYQPZCMPSA-\
vPb1uEpjb7HXSY4OP8UZiNkvw5kg
V/PushPlugin-NotificationService( 3847): GCM Registration webview: AGWebView -> URL: http://localhost/index.html uuid: 0f31c7a3-dd8a-4a88-b9fa-6107bea591c7
V/PushPlugin-NotificationService( 3847): handler found, sending registration id APA91bGN7FCQ2dfjMR757yAtn6fLiuVM0IeZmzwJKCUe6aNk9v7pIMXOXB16FPZFzME1r0ZrdPQYlpxLW8QzR2YepN2kv8WOiDW-kqLrmX\
Xjiaj-DqobjyzskFJzYQPZCMPSA-vPb1uEpjb7HXSY4OP8UZiNkvw5kgto handler: com.plugin.gcm.PushPlugin@b1312a30
V/PushPlugin( 3847): Registration Success called: APA91bGN7FCQ2dfjMR757yAtn6fLiuVM0IeZmzwJKCUe6aNk9v7pIMXOXB16FPZFzME1r0ZrdPQYlpxLW8QzR2YepN2kv8WOiDW-kqLrmXXjiaj-DqobjyzskFJzYQPZCMPSA-vP\
b1uEpjb7HXSY4OP8UZiNkvw5kg for instance WebView AGWebView -> URL: http://localhost/index.html uuid: 0f31c7a3-dd8a-4a88-b9fa-6107bea591c7 callBack PushPlugin27794453
D/CordovaPlugin( 3847): Sending plugin result to WebView: "APA91bGN7FCQ2dfjMR757yAtn6fLiuVM0IeZmzwJKCUe6aNk9v7pIMXOXB16FPZFzME1r0ZrdPQYlpxLW8QzR2YepN2kv8WOiDW-kqLrmXXjiaj-DqobjyzskFJzYQP\
ZCMPSA-vPb1uEpjb7HXSY4OP8UZiNkvw5kg"
D/AGWebView( 3847): Added plugin result to JS message queue: "APA91bGN7FCQ2dfjMR757yAtn6fLiuVM0IeZmzwJKCUe6aNk9v7pIMXOXB16FPZFzME1r0ZrdPQYlpxLW8QzR2YepN2kv8WOiDW-kqLrmXXjiaj-DqobjyzskFJz\
YQPZCMPSA-vPb1uEpjb7HXSY4OP8UZiNkvw5kg"
D/JsMessageQueue( 3847): Enqueued JsMessage: "APA91bGN7FCQ2dfjMR757yAtn6fLiuVM0IeZmzwJKCUe6aNk9v7pIMXOXB16FPZFzME1r0ZrdPQYlpxLW8QzR2YepN2kv8WOiDW-kqLrmXXjiaj-DqobjyzskFJzYQPZCMPSA-vPb1uE\
pjb7HXSY4OP8UZiNkvw5kg", paused: false
D/JsMessageQueue( 3847): Flushed 1 messages to JS side.
V/GCMBaseIntentService( 3847): Releasing wakelock
D/dalvikvm( 1276): GC_CONCURRENT freed 1536K, 46% free 6890K/12732K, paused 0ms+1ms, total 52ms
D/dalvikvm( 3847): GC_CONCURRENT freed 863K, 14% free 6212K/7208K, paused 26ms+1ms, total 29ms
D/ConnectivityService( 1276): handleInetConditionHoldEnd: net=0, condition=0, published condition=0
D/dalvikvm( 1276): GC_CONCURRENT freed 921K, 44% free 6870K/12176K, paused 30ms+1ms, total 37ms
```

**Comment by harsaharsa** [comment]
Ok, finally! We ended up rewriting the Cordova native bridge to directly evaluate the plugin results to the WebViews instead of using the online/offline event bridge, and it seems like it's working solidly. Huge apologies for the unreasonably long wait.

The fix is now available from the Build Service as 4.0.6-edge2, please give it a try and tell us if the issue persists!
